File: app/src/main/java/com/easy/kotlins/http/core/OkSource.java
package com.easy.kotlins.http.core;


import java.util.Objects;

/**
 * Create by LiZhanPing on 2020/4/29
 */
public class OkSource<T> {

    private T source;
    private Throwable error;

    private OkSource(T source) {
        this.source = source;
    }

    private OkSource(Throwable error) {
        this.error = error;
    }

    public static <T> OkSource<T> just(T source){
        return new OkSource<>(Objects.requireNonNull(source));
    }

    public static <T> OkSource<T> error(Throwable error){
        return new OkSource<>(error);
    }

    public T getSource() {
        return source;
    }

    public Throwable getError(){
        return error;
    }
}

File: common-ktx/src/main/java/com/easy/kotlins/http/core/OkSource.kt
package com.easy.kotlins.http.core


/**
 * Create by LiZhanPing on 2020/4/29
 */
class OkSource<T> {
    var source: T? = null
    var error: Throwable? = null
        private set

    private constructor(source: T) {
        this.source = source
    }

    private constructor(error: Throwable) {
        this.error = error
    }

    companion object {
        fun <T> just(source: T): OkSource<T> {
            return OkSource(requireNotNull(source))
        }

        fun <T> error(error: Throwable): OkSource<T> {
            return OkSource(error)
        }
    }
}